xbmc/pvr/timers/PVRTimers.cpp

```diff
@@ -450,7 +455,8 @@ bool CPVRTimers::GetSubDirectory(const CPVRTimersPath &path, CFileItemList &item
       if ((timer->m_bIsRadio == bRadio) &&
           (timer->m_iParentClientIndex != PVR_TIMER_NO_PARENT) &&
           (timer->m_iClientId == iClientId) &&
-          (timer->m_iParentClientIndex == iParentId))
+          (timer->m_iParentClientIndex == iParentId) &&
+          (!bHideDisabled || (timer->m_state != PVR_TIMER_STATE_DISABLED)))
```

Comment:
Really? Nah... too less duplicate code, imo. It's actually only <code>timer->m_bIsRadio == bRadio</code> and <code>!bHideDisabled || (timer->m_state != PVR_TIMER_STATE_DISABLED)</code>. What to make out of this? And how to name the new method doing just these two checks??